scIB/integration.py

```diff
@@ -283,49 +195,27 @@ def runBBKNN(adata, batch, hvg=None):
         corrected = bbknn.bbknn(adata, batch_key=batch, neighbors_within_batch=25, copy=True)
     return corrected
 
-def runConos(adata, batch, hvg=None):
-    checkSanity(adata, batch, hvg)
-    import time
-    import os
-    tmpName = "/tmp/conos"+time.time()
-    saveSeurat(adata, tmpName+'.RDS')
-    os.system('Rscript /home/icb/daniel.strobl/Benchmarking_data_integration/R/runConos.R '+tmpName+'.RDS '+batch+' '+tmpName)
-    adata_out = readConos(tmpName)
-
-    return adata_out
-
-    """anndata2ri.activate()
-    ro.r('library(Seurat)')
-    ro.r('library(scater)')
-    ro.r('library(conos)')
-
-    ro.globalenv['adata_c'] = adata
-    ro.r('sobj = as.Seurat(adata_c, counts = "counts", data = "X")')
-    ro.r(f'batch_list = SplitObject(sobj, split.by = "{batch}")')
-
-    ro.r('con <- Conos$new(batch_list)')
-    ro.r('con$buildGraph(k=15, k.self=5, space="PCA", ncomps=30)')
-    os.mkdir('conos_tmp')
-    ro.r('saveConosForScanPy(con, output.path="conos_tmp/", verbose=T))')
-
-    DATA_PATH = os.path.expanduser('conos_tmp/')
-    pca_df = pd.read_csv(DATA_PATH+'pca.csv')
-    
-    graph_conn_mtx = sp.io.mmread(DATA_PATH + "graph_connectivities.mtx")
-    graph_dist_mtx = sp.io.mmread(DATA_PATH + "graph_distances.mtx")
-    
-    out = adata.copy()
-    
-    out.X_pca = pca_df.values
-    
-    out.uns['neighbors'] = dict(connectivities=graph_conn_mtx.tocsr(), distances=graph_dist_mtx.tocsr())
-    
-    anndata2ri.deactivate()
-    return out
+
+def runSaucie(adata, batch):
     """
+    parametrisation from https://github.com/KrishnaswamyLab/SAUCIE/blob/master/scripts/SAUCIE.py
+    """
+    import SAUCIE
+    import sklearn.decomposition
+    pca_op = sklearn.decomposition.PCA(100)
+    expr = adata.X.todense()
+    data = pca_op.fit_transform(expr)
+    saucie = SAUCIE.SAUCIE(100, lambda_b=0.1)
+    loader_train = SAUCIE.Loader(data, labels=adata.obs[batch].cat.codes, shuffle=True)
+    loader_eval = SAUCIE.Loader(data, labels=adata.obs[batch].cat.codes, shuffle=False)
+    saucie.train(loader_train, steps=5000)
+    ret = adata.copy()
```

Comment:
quick question, why copy and not just add to the initial adata?

Reply:
I didn't want to overwrite adata.X

Reply:
good!